InsiteTeamTask/Repositories/DataRepository.cs

```diff
@@ -16,25 +16,39 @@ public List<Attendance> GetAttendanceListFor(int gameNumber)
             List<Member> members = service.Members().ToList();
             List<Ticket> tickets = service.Tickets().ToList();
             List<Product> products = service.Products().ToList();
+            List<Game> games = service.Games().ToList();
+            List<Season> seasons = service.Seasons().ToList();
 
             List<Attendance> attendanceList = new List<Attendance>();
 
-            for(int i = 0; i < members.Count; i++)
-            {
-                attendanceList.Add(new Attendance()
-                {
-                    Barcode = "N/A",
-                    MemberId = members[i].Id
-                });
-            }
+            //for (int i = 0; i < members.Count; i++)
+            //{
+            //    attendanceList.Add(new Attendance()
+            //    {
+            //        Barcode = "N/A",
+            //        MemberId = members[i].Id
+            //    });
+            //}
+
+            //for (int i = 0; i < tickets.Count; i++)
+            //{
+            //    attendanceList.Add(new Attendance()
+            //    {
+            //        Barcode = tickets[i].Barcode,
+            //        MemberId = 0
+            //    });
+            //}
 
-            for (int i = 0; i < tickets.Count; i++)
+            for (int i = 0; i < products.Count; i++)
             {
-                attendanceList.Add(new Attendance()
+                if (products[i].GameId == 3)
                 {
-                    Barcode = tickets[i].Barcode,
-                    MemberId = 0
-                });
+                    attendanceList.Add(new Attendance()
+                    {
+                        Barcode = tickets[i].Barcode,
```

Comment:
You're looping through `products` in this for loop, but indexing `tickets`. If the Product list is larger than the Ticket list, you'll get an error here for using an out of range index.

What I'd expect to see here is for you to retrieve a Game from the data using the `gameId` passed into this method, link this to a product using the Game's Id and SeasonId, and then retrieve all tickets with a matching ProductId.